jade-repl.el

```diff
@@ -220,6 +220,41 @@ LEVEL is a string representing the logging level, it can be
         ;; convenience
         (when error (message string))))))
 
+(defun jade-repl-emit-console-message-with-params (params level url line)
+  "Emit a console message."
+  (with-current-buffer (jade-repl-get-buffer)
+    (save-excursion
+      (goto-char jade-repl-output-end-marker)
+      (set-marker jade-repl-output-start-marker (point))
+      (jade-repl--emit-values params level url line)
+      (set-marker jade-repl-output-end-marker (point))
+      (unless (eolp)
+        (insert "\n")))))
+
+(defun jade-repl--emit-values (params level url line)
+  (if (= (seq-length params) 1)
+          (jade-repl--emit-single-value-message (seq-elt params 0) level url line)
+        (jade-repl--emit-multiple-values-message params level url line)))
+
+(defun jade-repl--emit-single-value-message (value level url line)
+  (insert "\n" level ": ")
+    (if (string= (map-elt value 'type) "string")
+        (insert (jade-description-string value) (jade-repl--format-url-line " " url line))
+      (progn (jade-render-value value)
+             (insert (jade-repl--format-url-line "\n" url line)))))
+
+(defun jade-repl--emit-multiple-values-message (values level url line)
+  (insert "\n" level ":")
+  (seq-do (lambda (value)
+            (insert "\n  ")
+            (jade-render-value value nil))
+          values)
+  (insert (jade-repl--format-url-line "\n" url line )))
+
+(defun jade-repl--format-url-line (whitespace url line)
+  (unless (string-empty-p url)
+      (format "%s%s:%s" whitespace url line)))
```

Comment:
Yes it's looks a bit ugly but I don't see a better way.
I don't want to insert `(Newline|Space)url:line` when it's something like `:1` e.g. when getting messages from the repl.

Another aproach will be to insert something like `repl` instead of `url:line` then I can get rid of this ugly param.